src/shipit_code_coverage/shipit_code_coverage/coverage_by_dir.py

```diff
@@ -41,7 +42,14 @@ def load_changesets(rev1, rev2):
 
 
 def get_related_bugs(changesets, directory):
-    return list(set([bug for bug, paths in changesets if any(fnmatch.fnmatch(path, directory + '/*') for path in paths)]))
+    bugs = set([bug for bug, paths in changesets if any(fnmatch.fnmatch(path, directory + '/*') for path in paths)])
```

Comment:
Nit: Can you make this return the list and then in `generate` create the dict for each bug?